### ISSUE TITLE
fresh upgrade-install   errors in error log

### ISSUE THREAD

**Issue by Kindred-999** [SimpleMachines/SMF#890]
When viewing an old post:
2: in_array() expects parameter 2 to be array, boolean given
http://test2.turtleshellprod.com/index.php?topic=6.0
File: /....../Sources/Display.php
Line: 1313

When posting a new thread:
 Type of error: Undefined
8: Undefined index: bbc_quote
http://test2.turtleshellprod.com/index.php?action=post;board=3.0
File: /home/turtleshell/www/test2/Themes/default/Post.template.php (main sub template - eval?)
Line: 714

 Type of error: Undefined
8: Undefined index: toggle_view
http://test2.turtleshellprod.com/index.php?action=post;board=3.0
File: /home/turtleshell/www/test2/Sources/Subs-Editor.php
Line: 1674

 Type of error: Undefined
8: Undefined index: unformat_text
http://test2.turtleshellprod.com/index.php?action=post;board=3.0
File: /home/turtleshell/www/test2/Sources/Subs-Editor.php
Line: 1670

 Type of error: Undefined
8: Undefined index: horizontal_rule
http://test2.turtleshellprod.com/index.php?action=post;board=3.0
File: /home/turtleshell/www/test2/Sources/Subs-Editor.php
Line: 1656

 Type of error: Undefined
8: Undefined index: list_ordered
http://test2.turtleshellprod.com/index.php?action=post;board=3.0
File: /home/turtleshell/www/test2/Sources/Subs-Editor.php
Line: 1652

 Type of error: Undefined
8: Undefined index: list_unordered
http://test2.turtleshellprod.com/index.php?action=post;board=3.0
File: /home/turtleshell/www/test2/Sources/Subs-Editor.php
Line: 1648

 Type of error: Undefined
8: Undefined index: bbc_quote
http://test2.turtleshellprod.com/index.php?action=post;board=3.0
File: /home/turtleshell/www/test2/Sources/Subs-Editor.php
Line: 1643

 Type of error: Undefined
8: Undefined index: bbc_code
http://test2.turtleshellprod.com/index.php?action=post;board=3.0
File: /home/turtleshell/www/test2/Sources/Subs-Editor.php
Line: 1639

 Type of error: Undefined
8: Undefined index: table
http://test2.turtleshellprod.com/index.php?action=post;board=3.0
File: /home/turtleshell/www/test2/Sources/Subs-Editor.php
Line: 1635

 Type of error: Undefined
8: Undefined index: teletype
http://test2.turtleshellprod.com/index.php?action=post;board=3.0
File: /home/turtleshell/www/test2/Sources/Subs-Editor.php
Line: 1630

 Type of error: Undefined
8: Undefined index: subscript
http://test2.turtleshellprod.com/index.php?action=post;board=3.0
File: /home/turtleshell/www/test2/Sources/Subs-Editor.php
Line: 1626

 Type of error: Undefined
8: Undefined index: superscript
http://test2.turtleshellprod.com/index.php?action=post;board=3.0
File: /home/turtleshell/www/test2/Sources/Subs-Editor.php
Line: 1622

 Type of error: Undefined
8: Undefined index: marquee
http://test2.turtleshellprod.com/index.php?action=post;board=3.0
File: /home/turtleshell/www/test2/Sources/Subs-Editor.php
Line: 1617

 Type of error: Undefined
8: Undefined index: shadow
http://test2.turtleshellprod.com/index.php?action=post;board=3.0
File: /home/turtleshell/www/test2/Sources/Subs-Editor.php
Line: 1613

 Type of error: Undefined
8: Undefined index: glow
http://test2.turtleshellprod.com/index.php?action=post;board=3.0
File: /home/turtleshell/www/test2/Sources/Subs-Editor.php
Line: 1609

**Comment by Oldiesmann** [comment]
All of those "undefined index" errors are missing language strings. Most of the editor buttons do have language strings in Editor.english.php, though not with the same indexes (sometimes it's just a different case - such as $txt['Bold'] instead of $txt['bold'], but there is no equivalent for "ordered_list" or "unordered_list").

**Comment by Kindred-999** [comment]
yup... I am aware that undefined indexes are usually text strings... but those should not be undefined in a baseline install... :)

**Comment by illori** [comment]
i am getting these errors when viewing a thread with the quick reply enabled.

**Comment by Arantor** [comment]
Yeah, we know what the problem is, we discussed this in IRC earlier. There are two difference issues here. The first error relates to the likes caching, and looks like it's been fixed.

What happened is recently, all the old editor's language strings were removed. The plan seems to have been to preserve all the definitions for the old editor even though the old editor is not properly invoked; the idea being to let mods have backwards compatibility. But as per all the recent discussions about backwards compatibility, it seems like the proper thing for us to do is drop this stuff (and fixing the error in the process by not invoking the old editor at all), and get mod authors to fix things up properly for 2.1.
